app/controllers/registrations_controller.rb

```diff
@@ -1,8 +1,15 @@
 class RegistrationsController < ApplicationController
+  def create
+    if current_user
+      Registration.create!(registration_params)
+    else
+      user = User.find_or_create_by(email: user_params[:email]) do |user|
+        user.name = user_params[:name]
```

Comment:
This isn't going to persist the user's name. Need to either user.save or user.update_attributes instead of user.name.

Reply:
Nevermind, I wasn't familiar with the block syntax.